frontend/src/lib/keybind.ts

```diff
@@ -1,29 +1,33 @@
-/**
- * Enum to represent the key strings provided by {@link KeyboardEvent}
- */
-export enum Keys {
-  J = 'j',
-  D = 'd',
-  K = 'k',
-  H = 'h',
-  V = 'v',
-  T = 't',
-  M = 'm',
-  L = 'l',
-  O = 'o',
-  S = 's',
-  SINGLE_QUOTE = "'",
-  FORWARD_SLASH = '/',
-  CTRL = 'Control',
-  ENTER = 'Enter',
-  ARROW_UP = 'ArrowUp',
-  ARROW_DOWN = 'ArrowDown',
-}
+import { match } from 'ts-pattern';
+import Result from '../routes/search/Result.svelte';
+import type Searchbar from './components/Searchbar.svelte';
+import type SpellCorrection from '../routes/search/SpellCorrection.svelte';
+
+export const Keys = [
+  'j',
+  'd',
+  'k',
+  'h',
+  'v',
+  't',
+  'm',
+  'l',
+  'o',
+  's',
+  "'",
+  '/',
+  'Control',
+  'Enter',
+  'ArrowUp',
+  'ArrowDown',
+];
+export type Key = (typeof Keys)[number];
 
 // The type used to declare the key downs that will trigger the callback
 export interface KeybindCallback {
-  key: Keys;
-  callback: (e: KeyboardEvent) => void;
+  key: Key;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  callback: (e: KeyboardEvent, context: any) => void;
```

Comment:
had a bit of a battle with typing here. I tried to create a generic and use it both here (ie. `context: Context`) and then again on line 39 with `export class Keybind<Context>` but everything I tried complained or didn't register that they would be the same type when implemented. totally open to any suggestions